src/mongocrypt-ctx-encrypt.c

```diff
@@ -447,21 +447,24 @@ get_command_name (_mongocrypt_buffer_t *cmd, mongocrypt_status_t *status)
 }
 
 static bool
-command_needs_deleteTokens (const char *command_name)
+command_needs_deleteTokens (mongocrypt_ctx_t *ctx, const char *command_name)
 {
-   const char *cmds_needing_deleteTokens[] = {
-      "delete", "update", "findAndModify"};
-
-   BSON_ASSERT_PARAM (command_name);
-
-   size_t i;
-   for (i = 0; i < sizeof (cmds_needing_deleteTokens) /
-                      sizeof (cmds_needing_deleteTokens[0]);
-        i++) {
-      if (0 == strcmp (cmds_needing_deleteTokens[i], command_name)) {
-         return true;
+   if (ctx->kb.crypt->opts.use_fle2_v2 == false) {
```

Comment:
Also suggest using `ctx->crypt` rather than `ctx->kb.crypt` to get the `mongocrypt_t` directly:

```suggestion
   if (ctx->crypt->opts.use_fle2_v2 == false) {
```

Reply:
Done.